mne/preprocessing/ica.py

```diff
@@ -842,9 +828,10 @@ def _transform_raw(self, raw, start, stop, reject_by_annotation=False):
             raise RuntimeError('No fit available. Please fit ICA.')
         start, stop = _check_start_stop(raw, start, stop)
 
-        picks = pick_types(raw.info, include=self.ch_names, exclude='bads',
-                           meg=False, ref_meg=False)
-        if len(picks) != len(self.ch_names):
+        picks = _picks_to_idx(
+            raw.info, self.ch_names, exclude=[], allow_empty=True)
+        raise_ = False if len(picks) == len(self.ch_names) else True
+        if raise_:
             raise RuntimeError('Raw doesn\'t match fitted data: %i channels '
                                'fitted but %i channels supplied. \nPlease '
                                'provide Raw compatible with '
```

Comment:
Looking at this code duplication (x3: raw, epochs, evoked), it's probably worth adding a `def _get_picks(self, inst, kind):` that has this picks, check, and raise logic and returns the picks to use from `inst` of kind `kind`.

Reply:
Done. I noticed it too, but didn't want to bother at the time with the different error messages depending on the instance.